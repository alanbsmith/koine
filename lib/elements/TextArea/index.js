import styled from 'styled-components';
import { rem } from 'polished';
import { applyStyleModifiers } from 'styled-components-modifiers';

import { focusColors } from '../../modifiers';

const TextArea = styled.textarea`
  background: none;
  border: none;
  border-bottom: 2px solid ${props => props.theme.colors.base.chrome100};
  box-sizing: border-box;
  flex: 1;
  font-size: ${rem(16)};
  outline: none;
  resize: vertical;
  transition: border-bottom 200ms ease;

  &:focus {
    border-bottom-color: ${props => props.theme.colors.base.chrome500};
  }

  ${applyStyleModifiers(focusColors)}
`;

export default TextArea;
