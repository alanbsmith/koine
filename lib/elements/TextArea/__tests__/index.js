import React from 'react';
import TextArea from '../index';
import 'jest-styled-components';

import { renderWithTheme } from '__tests__/helpers';


describe('TextArea Element', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(
      <TextArea modifiers={['']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the focusBlue modifier', () => {
    const tree = renderWithTheme(
      <TextArea modifiers={['focusBlue']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the focusGreen modifier', () => {
    const tree = renderWithTheme(
      <TextArea modifiers={['focusGreen']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the focusOrange modifier', () => {
    const tree = renderWithTheme(
      <TextArea modifiers={['focusOrange']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the focusPink modifier', () => {
    const tree = renderWithTheme(
      <TextArea modifiers={['focusPink']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the focusPurple modifier', () => {
    const tree = renderWithTheme(
      <TextArea modifiers={['focusPurple']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the focusYellow modifier', () => {
    const tree = renderWithTheme(
      <TextArea modifiers={['focusYellow']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the focusDanger modifier', () => {
    const tree = renderWithTheme(
      <TextArea modifiers={['focusDanger']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the focusDefault modifier', () => {
    const tree = renderWithTheme(
      <TextArea modifiers={['focusDefault']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the focusInfo modifier', () => {
    const tree = renderWithTheme(
      <TextArea modifiers={['focusInfo']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the focusSuccess modifier', () => {
    const tree = renderWithTheme(
      <TextArea modifiers={['focusSuccess']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds the focusWarning modifier', () => {
    const tree = renderWithTheme(
      <TextArea modifiers={['focusWarning']} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
