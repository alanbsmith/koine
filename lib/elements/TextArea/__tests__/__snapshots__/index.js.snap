// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Element adds the focusBlue modifier 1`] = `
.c0 {
  background: none;
  border: none;
  border-bottom: 2px solid #F9FAFC;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.25rem;
  outline: none;
  resize: vertical;
  -webkit-transition: border-bottom 200ms ease;
  transition: border-bottom 200ms ease;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

<textarea
  className="c0"
/>
`;

exports[`TextArea Element adds the focusDanger modifier 1`] = `
.c0 {
  background: none;
  border: none;
  border-bottom: 2px solid #F9FAFC;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.25rem;
  outline: none;
  resize: vertical;
  -webkit-transition: border-bottom 200ms ease;
  transition: border-bottom 200ms ease;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

.c0:focus {
  border-bottom-color: #cc1111;
}

<textarea
  className="c0"
/>
`;

exports[`TextArea Element adds the focusDefault modifier 1`] = `
.c0 {
  background: none;
  border: none;
  border-bottom: 2px solid #F9FAFC;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.25rem;
  outline: none;
  resize: vertical;
  -webkit-transition: border-bottom 200ms ease;
  transition: border-bottom 200ms ease;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

.c0:focus {
  border-bottom-color: #8492A6;
}

<textarea
  className="c0"
/>
`;

exports[`TextArea Element adds the focusGreen modifier 1`] = `
.c0 {
  background: none;
  border: none;
  border-bottom: 2px solid #F9FAFC;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.25rem;
  outline: none;
  resize: vertical;
  -webkit-transition: border-bottom 200ms ease;
  transition: border-bottom 200ms ease;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

.c0:focus {
  border-bottom-color: #13CE66;
}

<textarea
  className="c0"
/>
`;

exports[`TextArea Element adds the focusInfo modifier 1`] = `
.c0 {
  background: none;
  border: none;
  border-bottom: 2px solid #F9FAFC;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.25rem;
  outline: none;
  resize: vertical;
  -webkit-transition: border-bottom 200ms ease;
  transition: border-bottom 200ms ease;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

.c0:focus {
  border-bottom-color: #0492d2;
}

<textarea
  className="c0"
/>
`;

exports[`TextArea Element adds the focusOrange modifier 1`] = `
.c0 {
  background: none;
  border: none;
  border-bottom: 2px solid #F9FAFC;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.25rem;
  outline: none;
  resize: vertical;
  -webkit-transition: border-bottom 200ms ease;
  transition: border-bottom 200ms ease;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

.c0:focus {
  border-bottom-color: #FF7849;
}

<textarea
  className="c0"
/>
`;

exports[`TextArea Element adds the focusPink modifier 1`] = `
.c0 {
  background: none;
  border: none;
  border-bottom: 2px solid #F9FAFC;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.25rem;
  outline: none;
  resize: vertical;
  -webkit-transition: border-bottom 200ms ease;
  transition: border-bottom 200ms ease;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

.c0:focus {
  border-bottom-color: #FF49DB;
}

<textarea
  className="c0"
/>
`;

exports[`TextArea Element adds the focusPurple modifier 1`] = `
.c0 {
  background: none;
  border: none;
  border-bottom: 2px solid #F9FAFC;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.25rem;
  outline: none;
  resize: vertical;
  -webkit-transition: border-bottom 200ms ease;
  transition: border-bottom 200ms ease;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

.c0:focus {
  border-bottom-color: #7E5BEF;
}

<textarea
  className="c0"
/>
`;

exports[`TextArea Element adds the focusSuccess modifier 1`] = `
.c0 {
  background: none;
  border: none;
  border-bottom: 2px solid #F9FAFC;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.25rem;
  outline: none;
  resize: vertical;
  -webkit-transition: border-bottom 200ms ease;
  transition: border-bottom 200ms ease;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

.c0:focus {
  border-bottom-color: #00a651;
}

<textarea
  className="c0"
/>
`;

exports[`TextArea Element adds the focusWarning modifier 1`] = `
.c0 {
  background: none;
  border: none;
  border-bottom: 2px solid #F9FAFC;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.25rem;
  outline: none;
  resize: vertical;
  -webkit-transition: border-bottom 200ms ease;
  transition: border-bottom 200ms ease;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

.c0:focus {
  border-bottom-color: #f2c144;
}

<textarea
  className="c0"
/>
`;

exports[`TextArea Element adds the focusYellow modifier 1`] = `
.c0 {
  background: none;
  border: none;
  border-bottom: 2px solid #F9FAFC;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.25rem;
  outline: none;
  resize: vertical;
  -webkit-transition: border-bottom 200ms ease;
  transition: border-bottom 200ms ease;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

.c0:focus {
  border-bottom-color: #FFC82C;
}

<textarea
  className="c0"
/>
`;

exports[`TextArea Element renders correctly 1`] = `
.c0 {
  background: none;
  border: none;
  border-bottom: 2px solid #F9FAFC;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1.25rem;
  outline: none;
  resize: vertical;
  -webkit-transition: border-bottom 200ms ease;
  transition: border-bottom 200ms ease;
}

.c0:focus {
  border-bottom-color: #1FB6FF;
}

<textarea
  className="c0"
/>
`;
